### ISSUE TITLE
Error reading tfstate file: 0.12 format error: <nil>; pre-0.12 format error: <nil> (nil error means no content/modules found in the respective format)

### ISSUE THREAD

**Issue by queglay** [adammck/terraform-inventory#121]
I updated to tf .12.6 and terraform inventory .9, but I'm getting errors that the tfstate  is not current.  I'm not sure why, because this is a whole new rollout started from scratch with no lingering .11 to my knowledge.

```
vagrant@openfirehawkserverdev:/vagrant$ /vagrant/ansible/inventory/terraform-inventory --list
Error reading tfstate file: 0.12 format error: <nil>; pre-0.12 format error: <nil> (nil error means no content/modules found in the respective format)
vagrant@openfirehawkserverdev:/vagrant$ /vagrant/ansible/inventory/terraform-inventory --version
/vagrant/ansible/inventory/terraform-inventory version 0.9
vagrant@openfirehawkserverdev:/vagrant$ terraform --version
Terraform v0.12.6
+ provider.aws v2.24.0
+ provider.null v2.1.2
+ provider.random v2.2.0
vagrant@openfirehawkserverdev:/vagrant$ 
```

**Comment by jimb0bmij** [comment]
> I updated to tf .12.6 and terraform inventory .9, but I'm getting errors that the tfstate is not current. I'm not sure why, because this is a whole new rollout started from scratch with no lingering .11 to my knowledge.
> 
> ```
> vagrant@openfirehawkserverdev:/vagrant$ /vagrant/ansible/inventory/terraform-inventory --list
> Error reading tfstate file: 0.12 format error: <nil>; pre-0.12 format error: <nil> (nil error means no content/modules found in the respective format)
> vagrant@openfirehawkserverdev:/vagrant$ /vagrant/ansible/inventory/terraform-inventory --version
> /vagrant/ansible/inventory/terraform-inventory version 0.9
> vagrant@openfirehawkserverdev:/vagrant$ terraform --version
> Terraform v0.12.6
> + provider.aws v2.24.0
> + provider.null v2.1.2
> + provider.random v2.2.0
> ```

Getting the same issue but using provider.vsphere

**Comment by tbennett6421** [comment]
You can try `terraform-inventory --list ./`; that worked for me.

**Comment by tbennett6421** [comment]
#114 see PR for more details

**Comment by BenTheRabbit** [comment]
> You can try `terraform-inventory --list ./`; that worked for me.

Indeed, but as you cannot use arguments with ansible, you cannot call it from the --inventory-file option. Or am I doing something wrong?

```
TF_STATE=/path/to/terraform.tfstate ansible-playbook --inventory-file=/path/to/terraform-inventory test.yml
 [WARNING]:  * Failed to parse /path/to/terraform-inventory with script plugin: Inventory script (/path/to/terraform-inventory) had an execution error: Error reading
tfstate file: 0.12 format error: <nil>; pre-0.12 format error: <nil> (nil error means no content/modules found in the respective format)
```

**Comment by BenTheRabbit** [comment]
> > You can try `terraform-inventory --list ./`; that worked for me.
> 
> Indeed, but as you cannot use arguments with ansible, you cannot call it from the --inventory-file option. Or am I doing something wrong?
> 
> ```
> TF_STATE=/path/to/terraform.tfstate ansible-playbook --inventory-file=/path/to/terraform-inventory test.yml
>  [WARNING]:  * Failed to parse /path/to/terraform-inventory with script plugin: Inventory script (/path/to/terraform-inventory) had an execution error: Error reading
> tfstate file: 0.12 format error: <nil>; pre-0.12 format error: <nil> (nil error means no content/modules found in the respective format)
> ```

ok, works if `TF_STATE=./`

**Comment by alor** [comment]
any plan to fix this without forcing the user to add the TF_STATE variable or the ./ at the end of the command?  thank you

**Comment by augustgerro** [comment]
Does not works even with `./`
<img width="711" alt="image" src="https://user-images.githubusercontent.com/3669104/76459749-cb740a00-63ed-11ea-9b02-1cb0be395466.png">

**Comment by DaFresh** [comment]
+1
Same result here.

Terraform v0.12.25
+ provider.openstack v1.28.0

**Comment by a0s** [comment]
Working inventory:

ansible/inventory/a_terraform.sh: 
```shell
#!/usr/bin/env bash
CURRENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
TERRAFORM_INVENTORY=`which terraform-inventory`
TF_STATE="$CURRENT_DIR/../../terraform"    # <<== relative path to dir with .tfstate
"$TERRAFORM_INVENTORY" "$@" "$TF_STATE"
```
